pkg/chaosdaemon/jvm_server_backup.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package chaosdaemon
+
+import (
+	"context"
+	"fmt"
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/util"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+	pb "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+const (
+	bytemanHome            = "/usr/local/byteman/"
+	bmInstallCommandBackUp = bytemanHome + "bin/bminstall.sh -b -Dorg.jboss.byteman.transform.all -Dorg.jboss.byteman.verbose -Dorg.jboss.byteman.compile.to.bytecode -p %d %d"
+	bmSubmitCommandBackUp  = bytemanHome + "bin/bmsubmit.sh -p %d -%s %s"
+)
+
+func (s *DaemonServer) InstallJVMRulesBackUp(ctx context.Context,
+	req *pb.InstallJVMRulesRequest) (*empty.Empty, error) {
+	pid, err := s.crClient.GetPidFromContainerID(ctx, req.ContainerId)
+	if err != nil {
+		log.Error(err, "GetPidFromContainerID")
+		return nil, err
+	}
+
+	//todo get pid in the container，
+	processBuilder := bpm.DefaultProcessBuilder("sh", "-c", "ps -ef | grep java | grep -v grep | awk 'NR==1 {print $2}'").SetContext(ctx).SetNS(pid, bpm.MountNS).SetNS(pid, bpm.PidNS)
```

Comment:
The result of this script is not trustable. For example, the return value of this script in `examples/jvm/app.yaml` is `root`:

```
chaos-daemon-server     get java pid    {"output": "root\n"}
```

I wonder why we cannot suggest the first process is the java process?

Reply:
@zhqqqy I removed this logic in my latest commit. If you think it's not reasonable, feel free to add it back with a comment of the situation when the java process is not the first one.

Reply:
> The result of this script is not trustable. For example, the return value of this script in `examples/jvm/app.yaml` is `root`:
> 
> ```
> chaos-daemon-server     get java pid    {"output": "root\n"}
> ```
> 
> I wonder why we cannot assume the first process is the java process?

Because in our internal system, the process mounted by the container is an environment initialization script. When the container starts, some necessary programs, such as `databus, crond, nscd`, etc., will be initialized first, and then the java project will be started.

Reply:
how about this?
```
ps -eo pid,comm | grep java | awk 'NR==1 {print $1}'
```